components/Books.js

```diff
@@ -295,7 +298,7 @@ class Books extends React.Component {
               Internet Archive
             </span>
           </div>
-          <form onSubmit={this.onSubmit}>
+          <form onSubmit={(e) => this.onSubmit(e, session.user.name)}>
```

Comment:
@Parthiv-M Please have a defensive check here to handle exceptions in case this is not found.